lib/model/QubitDigitalObject.php

```diff
@@ -1929,38 +1935,31 @@ public function createRepresentations($usageId, $connection = null)
     /**
      * Set 'page_count' property for this asset.
      *
-     * NOTE: requires the ImageMagick library
+     * Requires the imagick extension.
      *
      * @param null|mixed $connection
      *
      * @return QubitDigitalObject this object
      */
     public function setPageCount($connection = null)
     {
-        if ($this->canThumbnail() && sfImageMagickAdapter::isImageMagickAvailable()) {
-            $filename = ($this->derivativesGeneratedFromExternalMaster($this->usageId)) ? $this->getLocalPath() : $this->getAbsolutePath();
+        if (!$this->canThumbnail()) {
+            return $this;
+        }
 
-            $extension = pathinfo($filename, PATHINFO_EXTENSION);
+        $filename = ($this->derivativesGeneratedFromExternalMaster($this->usageId)) ? $this->getLocalPath() : $this->getAbsolutePath();
 
-            // If processing a PDF, attempt to use pdfinfo as it's faster
-            if ('pdf' == strtolower($extension) && sfImageMagickAdapter::pdfinfoToolAvailable()) {
-                $pages = sfImageMagickAdapter::getPdfinfoPageCount($filename);
-            } else {
-                $command = 'identify '.$filename;
-                exec($command, $output, $status);
-                $pages = count($output);
-            }
+        $im = new Imagick();
```

Comment:
We should check that `extension_loaded('imagick')` is true before and issue some kind of error if it is not.